### DISCUSSION TITLE
Port is already in use error when running tests

### DISCUSSION THREAD

**Discussion by yevhendevbox** [vercel/turborepo#7318] [Help] [answered]
Hey there, trying to configure monorepo using Turborepo first time. When running tests using turbo (in all workspaces) i get an error in the console: 

[error] WebSocket server error: Port is already in use

Workspaces using Vitest for testing, i guess it's because turbo running script in parallel, so they run on the same port or some thing. 

Any tips how to fix this? Thanks for any help rly.

**Comment by anthonyshew** [comment, marked as answer]
Because you're running multiple processes in parallel, yes, some processes could try to run on the same ports as each other. Most modern tooling will search for an open port rather, though, unless you've explicitly set a specific one.

Do you have a repo you could share? Someone may be able to take a look.
